synapseclient/extensions/curator/schema_generation.py

```diff
@@ -3998,6 +4061,10 @@ def _generate_jsonschema(
         return generator.component_json_schema, str(generator.output_path)
 
 
+@deprecated(
+    version="4.11.0",
```

Comment:
Not entirely sure if 4.11 is the right version. Put it as a placeholder for now.